echo-pubsub/src/main/java/com/netflix/spinnaker/echo/config/amazon/AmazonPubsubProperties.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.config.amazon;
+
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+import org.hibernate.validator.constraints.NotEmpty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import javax.validation.Valid;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+@Data
+@ConfigurationProperties(prefix = "pubsub.amazon")
+public class AmazonPubsubProperties {
+
+  @Valid
+  private List<AmazonPubsubSubscription> subscriptions;
+
+  @Data
+  public static class AmazonPubsubSubscription {
+
+    private static final Logger log = LoggerFactory.getLogger(AmazonPubsubSubscription.class);
+
+    @NotEmpty
+    String name;
+
+    @NotEmpty
+    String accountName;
+
+    @NotEmpty
+    String topicARN;
+
+    @NotEmpty
+    String queueARN;
+
+    String templatePath;
+
+    MessageFormat messageFormat;
+
+    int visibilityTimeout = 30;
+    int sqsMessageRetentionPeriodSeconds = 120;
+    int waitTimeSeconds = 5;
+
+    public AmazonPubsubSubscription() {
+    }
+
+    public AmazonPubsubSubscription(
```

Comment:
I find Lombok more opaque than just generating the constructors.

Reply:
Quite like the `echo-pubsub-aws` route, so we're not forced into the javas.

Reply:
I'll take the `echo-pubsub-aws` route.

Reply:
👍

Reply:
We're pretty far along reviewing this for a large refactor like moving this into a separate submodule and rewriting in a different language, particularly in a language I'm unfamiliar with. Echo currently has no Kotlin in it, so I'm also hesitant to introduce another language.

I would be happy to discuss any refactoring ideas you guys have, but I want to get this module functionally complete before making larger scale changes.

Reply:
@jtk54 when I pull it to a different module I'll keep it in Java. I'll do that in a separate PR, this one has had a lot of review. But, for data classes, Kotlin is quite nice. We are trying to gradually shift to Kotlin as the language of choice. First step to that is to not add any more Groovy code because the two don't mix well.

We can chat more offline about why we like the idea of pulling the aws bits to a new module, the google bits to a new module, and making the core pubsub functionality one module.